deployer: add info generating commands under resource-allocation

consideRatio opened the pull request:
This PR adds two commands of relevance towards refining the resource-allocation script to meet more needs. The added commands are updating their separate .yaml file's with collected information.

- Closes #3338 
- Closes #3330 

## Script overview

- `daemonset-requests` summarizes the requests from all daemonsets, and clarifies what daemonsets contributed to the requests and what daemonset didn't.

  The script makes us better understand and clarify differences observed. It helped me verify that the current overhead is mostly cloud provoider specific, to some extent cluster specific (feature and k8s version differences), and to a small extent instance type/node pool specific (gpu drivers).
  
  A caveat of this implementation strategy is that it doesn't intelligently rule out that a daemonset may not only be schedule on some nodes, for example core nodes only, or on nodes with gpus. I've filtered out the gpu related daemonsets manually, and not observed any core node pool only daemonsets.
- `instance-capacities` scrapes the `kubectl get node` reported `capacity` and `allocatable` (allocatable capacity), and saves both the lowest and highest reported value to help us overview the spread of these values across clusters. It turns out that there is a slight spread, but its very low.

## With `update-node-info`, why these commands?

These are iterations on `update-node-info`, but they are added separately as `update-node-info` directly hooks into the `resource-allocation choices` command, making an update to it break the existing behavior.

GeorgianaElena commented on deployer/README.md:
```diff
@@ -264,6 +268,12 @@ This sub-command can be used to generate the resource allocation choices for giv
 This generates a custom number of resource allocation choices for a certain instance type, depending on a certain chosen strategy that can be used in the profile list of a hub.
 
+##### `generate resource-allocation daemonset-overhead`
+Updates `daemonset_overhead.yaml` with an individual cluster's DaemonSets with running pods combined requests of CPU and memory, excluding GPU related DaemonSets.
```
This is currently a bit hard to understand. Maybe expand it a bit more with info from the PR's top comment?

```suggestion
This script updates information about a specific cluster in the `daemonset_overhead.yaml` file.

This file contains information about cpu/memory requests made by
DaemonSets with running pods in _all_ of our clusters but excluding GPU related DaemonSets.
```

consideRatio replied:
The command will not collect information from each cluster. I considered doing that, but I don't think the complexity added to `deployer` is justified when we can easily invoke it for all clusters one at the time with a command like `ls config/clusters | xargs -I {} deployer generate resource-allocation daemonset-overhead {}`.

I'll mention that in the docs here!

GeorgianaElena commented on deployer/README.md:
```diff
@@ -264,6 +268,12 @@ This sub-command can be used to generate the resource allocation choices for giv
+Updates `daemonset_overhead.yaml` with an individual cluster's DaemonSets with running pods combined requests of CPU and memory, excluding GPU related DaemonSets.
+
+##### `generate resource-allocation instance-capacity`
+Updates `instance_capacity.yaml` with instance types' total and allocatable capacity reported by `kubectl get node`.
```
From what I understand this command requires a cluster, and it uses that cluster to get the running nodes and updates the file with info about that or those running nodes, right?

From this description I understand that it updates all instance types and I wouldn't expect the command to require a cluster.

So maybe rephrase to:
```suggestion
This script updates the information in the `instance_capacity.yaml` file about the instance types found running in a given cluster.

This file contains the total and allocatable capacity as reported by `kubectl get node` of the instance types'  used across our clusters.
```

consideRatio replied:
I'll add a commit to refine this part similar to your suggestion, both here and in the .py file where this text is also duplicated as a docstring!

GeorgianaElena commented on deployer/commands/generate/resource_allocation/daemonset_overhead.py:
```diff
@@ -0,0 +1,152 @@
+        return "gke", server_version_info
+    if "eks" in server_version_info:
+        return "eks", server_version_info
+    else:
+        return "aks", server_version_info
```
```suggestion
    return "aks", server_version_info
```

GeorgianaElena commented on deployer/commands/generate/resource_allocation/daemonset_overhead.py:
```diff
@@ -0,0 +1,152 @@
+
+def get_daemon_sets_requests():
+    """
+    Returns a list of info about DaemonSets with pods desired to be scheduled on
```
```suggestion
    Returns a list of dicts with info about DaemonSets with pods desired to be scheduled on
```

GeorgianaElena commented on deployer/commands/generate/resource_allocation/daemonset_overhead.yaml:
```diff
@@ -0,0 +1,180 @@
+# - We may deploy or change a DaemonSet's requests over time (support-cryptnono,
+#   support-prometheus-node-exporter)
+#
+# This file isn't updated by automation, but can easily be updated by manually
+# by running a command once for each cluster:
```
```suggestion
# This file isn't updated by automation, but can easily be updated by manually
# running a command once for each cluster:
```

GeorgianaElena reviewed (approved):
🚀 Thanks @consideRatio

consideRatio commented:
Thank you for reviewing this @GeorgianaElena! I've not resolved all comments, but will get it done soonish!

consideRatio commented:
- Rebased
- renamed command instance-capacity to instance-capacities
- renamed command daemonset-overhead to daemonset-requests
- updated docstrings to reflect feedback from @GeorgianaElena

---

I'll go for a merge here at this point!
